From 450b7fa4b494c16eaf174689e996dc53aa4ecfec Mon Sep 17 00:00:00 2001
From: Guy Morand <guy.morand@bytesatwork.ch>
Date: Mon, 17 Jun 2024 08:57:00 +0200
Subject: [PATCH] arm64: freescale: imx93-byteengine.dts: Adapt tca6418

Add required pin muxing for the interrupt line and interrupt parent for
interrupt handling.

Signed-off-by: Guy Morand <guy.morand@bytesatwork.ch>
---
 arch/arm64/boot/dts/freescale/imx93-byteengine.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx93-byteengine.dtsi b/arch/arm64/boot/dts/freescale/imx93-byteengine.dtsi
index 06e58094f22b..666cef71cc82 100644
--- a/arch/arm64/boot/dts/freescale/imx93-byteengine.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx93-byteengine.dtsi
@@ -113,6 +113,8 @@ tca6418e: tca6418e@34 {
 		reg = <0x34>;
 		gpio-controller;
 		#gpio-cells = <2>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <10 IRQ_TYPE_EDGE_FALLING>;
 	};
 };
 
@@ -186,6 +188,7 @@ pinctrl_lpi2c2: lpi2c2grp {
 		fsl,pins = <
 			MX93_PAD_I2C2_SCL__LPI2C2_SCL		0x40000b9e
 			MX93_PAD_I2C2_SDA__LPI2C2_SDA		0x40000b9e
+			MX93_PAD_PDM_BIT_STREAM1__GPIO1_IO10	0x1280
 		>;
 	};
 
